src/components/NewWorkspaceModal.js

```diff
@@ -32,48 +30,41 @@ const constraints = {
   }
 }
 
-export default _.flow(
-  Utils.withCancellationSignal,
-  Utils.connectStore(authStore, 'authState')
-)(class NewWorkspaceModal extends Component {
-  static propTypes = {
-    cloneWorkspace: PropTypes.object,
-    onDismiss: PropTypes.func.isRequired,
-    customMessage: PropTypes.string,
-    title: PropTypes.string,
-    buttonText: PropTypes.string
-  }
+const NewWorkspaceModal = Utils.withDisplayName('NewWorkspaceModal', ({
+  cloneWorkspace, onSuccess, onDismiss, customMessage, requiredAuthDomain, title, buttonText
```

Comment:
@Shakespeared and I want to know if there is no propType equivalent in hook type components?

Reply:
So there actually is, we could put this below the declaration:

```js
NewWorkspaceModal.propTypes = {
  cloneWorkspace: PropTypes.object,
  ...
```

I judged that it wasn't worth it in this case, but I'm open to discussion.

Reply:
I'm wary of not requiring an `onDismiss` function- what would be the default behavior without one?